Framework/Muon/src/ConvertFitFunctionForMuonTFAsymmetry.cpp

```diff
@@ -282,19 +303,31 @@ IFunction_sptr ConvertFitFunctionForMuonTFAsymmetry::extractUserFunction(
 std::vector<double> ConvertFitFunctionForMuonTFAsymmetry::getNorms() {
   API::ITableWorkspace_sptr table = getProperty("NormalizationTable");
   const std::vector<std::string> wsNames = getProperty("WorkspaceList");
-
   std::vector<double> norms(wsNames.size(), 0);
-  auto colNames = table->getColumnNames();
-  auto wsNamesIndex = findName(colNames, "name");
-  auto normIndex = findName(colNames, "norm");
 
-  for (size_t row = 0; row < table->rowCount(); row++) {
+  if (table) {
+    auto colNames = table->getColumnNames();
+    auto wsNamesIndex = findName(colNames, "name");
+    auto normIndex = findName(colNames, "norm");
+
+    for (size_t row = 0; row < table->rowCount(); row++) {
+      for (size_t wsPosition = 0; wsPosition < wsNames.size(); wsPosition++) {
+        std::string wsName = wsNames[wsPosition];
+        std::replace(wsName.begin(), wsName.end(), ' ', ';');
+        if (table->String(row, wsNamesIndex) == wsName) {
+          norms[wsPosition] = table->Double(row, normIndex);
+        }
+      }
+    }
+  } else {
     for (size_t wsPosition = 0; wsPosition < wsNames.size(); wsPosition++) {
       std::string wsName = wsNames[wsPosition];
-      std::replace(wsName.begin(), wsName.end(), ' ', ';');
-      if (table->String(row, wsNamesIndex) == wsName) {
-        norms[wsPosition] = table->Double(row, normIndex);
-      }
+      API::MatrixWorkspace_const_sptr ws =
+          API::AnalysisDataService::Instance().retrieveWS<API::MatrixWorkspace>(
+              wsName);
+      const Mantid::API::Run &run = ws->run();
+      norms[wsPosition] =
+          std::stod(run.getProperty("analysis_asymmetry_norm")->value());
```

Comment:
Do we want to add a check here to catch the case where a normalization table is not provided but the workspaces do not contain the expected log?

Reply:
this should catch it https://github.com/mantidproject/mantid/pull/25890/files#diff-6ad3c3deb53abf4c25903e6a3c2fc229R165